lib/ansible/modules/cloud/azure/azure_rm_mysqlserver.py

```diff
@@ -0,0 +1,368 @@
+    sample: id
+version:
+    description:
+        - "Server version. Possible values include: '5.6', '5.7'"
+    returned: always
+    type: str
+    sample: version
+user_visible_state:
+    description:
+        - "A state of a server that is visible to user. Possible values include: 'Ready', 'Dropping', 'Disabled'"
+    returned: always
+    type: str
+    sample: user_visible_state
+fully_qualified_domain_name:
+    description:
+        - The fully qualified domain name of a server.
+    returned: always
+    type: str
+    sample: fully_qualified_domain_name
+'''
+
+import time
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.rdbms.mysql import MySQLManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMServers(AzureRMModuleBase):
+    """Configuration class for an Azure RM MySQL Server resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            sku=dict(
+                type='dict',
+                required=False
+            ),
+            location=dict(
+                type='str',
+                required=False
+            ),
+            storage_mb=dict(
+                type='long',
+                required=False
+            ),
+            version=dict(
+                type='str',
+                required=False
+            ),
+            ssl_enforcement=dict(
+                type='str',
+                required=False
+            ),
+            create_mode=dict(
+                type='str',
+                default='Default'
+            ),
+            admin_username=dict(
+                type='str',
+                required=False
+            ),
+            admin_password=dict(
+                type='str',
+                no_log=True,
+                required=False
+            ),
+            state=dict(
+                type='str',
+                required=False,
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.parameters = dict()
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMServers, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
```

Comment:
I'm curious as to whether you actually support tags in this, I don't see code that handles it.